modules/library/src/main/scala/zio/elasticsearch/ElasticSort.scala

```diff
@@ -40,4 +40,14 @@ object ElasticSort {
       order = None,
       unmappedType = None
     )
+
+  def sortByScript(script: String, sourceType: SourceType): SortByScript =
```

Comment:
I would create `Script` as `AnyVal` or `Newtype`, and use `sortBy` here as well. 

```
sortBy(field: String, ...)
sortBy(script: Script, ...)
```